src/back/msl/writer.rs

```diff
@@ -1721,9 +1721,56 @@ impl<W: Write> Writer<W> {
                     self.put_expression(arg, context, true)?;
                     write!(self.out, ") + 1) % 33) - 1)")?;
                 } else if fun == Mf::FindMsb {
-                    write!(self.out, "((({NAMESPACE}::clz(")?;
-                    self.put_expression(arg, context, true)?;
-                    write!(self.out, ") + 1) % 33) - 1)")?
+                    let inner = context.resolve_type(arg);
+
+                    if let Some(crate::ScalarKind::Uint) = inner.scalar_kind() {
+                        write!(self.out, "((({NAMESPACE}::clz(")?;
```

Comment:
Not your bug, but this looks really wrong; filed as #2236.

Reply:
Okay, I've simplified things, and hopefully addressed this issue.